lib/BackbeatConsumer.js

```diff
@@ -115,14 +116,32 @@ class BackbeatConsumer extends EventEmitter {
                         const bytes = locations.reduce((sum, item) =>
                             sum + item.size, 0);
 
-                        if (!this._metricsStore[qEntry.bucket]) {
-                            this._metricsStore[qEntry.bucket] = {
+                        const repInfo = qEntry.getReplicationInfo();
+                        let site;
+                        if (repInfo.storageType) {
+                            // known external cloud (i.e. aws_s3 || azure)
+                            site = repInfo.storageType;
+                        } else if (repInfo.storageClass) {
+                            site = repInfo.storageClass;
+                        } else {
+                            const id = uuid().replace(/-/g, '');
+                            site = `bb-unknown-type-${id}`;
```

Comment:
Just to note, I don't ever expect an unknown type to happen. It is here more as a precautionary measure and we will see the logs and may be able to debug from there.